src/methods/query-sync.ts

```diff
@@ -0,0 +1,91 @@
+import { Clock, combine, createStore, sample, Store, Unit } from 'effector';
+
+import { RouteInstance, RouteQuery } from '../types';
+import { createRouterControls } from './create-router-controls';
+
+type QueryCleanupStrategy = {
+  irrelevant: boolean;
+  empty: boolean;
+  preserve: string[];
+};
+
+type QuerySyncParams<T extends Record<string, Store<any>>> = {
+  source: T;
+  clock?: Clock<any>;
+  controls: ReturnType<typeof createRouterControls>;
```

Comment:
Is the `controls` parameter must be required?

Reply:
Yes, otherwise we can't access the router

Reply:
But there is no `controls` parameter in your examples. 🤔

Reply:
Oops, forgot about them